### PR TITLE
Run pyupgrade on the sources

### PR DESCRIPTION
**Pull request by sobolevn**
This PR introduces quite a lot of changes, please feel free to close if this creates unneeded churn.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -1,4 +1,3 @@
-# -*- coding: utf-8 -*-
 #
```
Comment (AlexWaygood) [inline docs/source/conf.py:1]:
```suggestion
```

**Comment by AlexWaygood** [comment]
Oh I thought the mypyc-segfaults-on-namedtuples-with-forward-refs issue had been fixed by https://github.com/python/mypy/pull/14401 (cc. @ichard26)

**Comment by ichard26** [comment]
> Edit: I guess this is a slightly different situation, since it specifically involves __future__ annotations rather than forward refs in general?

Yeah, PR #14401 only fixes stringified forward references. Non-stringified forward references (thanks to `from __future__ import annotations`) are still liable to blowing up. I can take a look this weekend at fixing that.

**Comment by sobolevn** [comment]
@ichard26 please ping me afterwards to merge this :)

**Pull request closed without merging**